Update project to require C++17

OsamaRab3 opened the pull request:
enforce -std=c++17 support using AX_CXX_COMPILE_STDCXX_17

OsamaRab3 commented:
@antoninbas please review my PR

antoninbas commented on targets/simple_switch_grpc/m4/ax_cxx_compile_stdcxx_17.m4:
```diff
@@ -0,0 +1,35 @@
+# =============================================================================
```
I think you also need https://www.gnu.org/software/autoconf-archive/ax_cxx_compile_stdcxx.html in this m4 directory?

antoninbas commented on m4/ax_cxx_compile_stdcxx.m4:
```diff
@@ -0,0 +1,1018 @@
+
+#endif  // __cplusplus < 202002L && !defined _MSC_VER
+
+]])
```
file is missing a trailing newline character (applies to all the other new files as well)

antoninbas reviewed (commented):
I wonder if the reference to C++11 in the README should be updated?

I see multiple CI failures for your PR.

OsamaRab3 commented:
> I see multiple CI failures for your PR.

I'm not sure about the exact cause of the CI failures at the moment. Could you please provide more details about the specific failures? I'd be happy to investigate further and work on resolving these issues. Any additional information or logs you can share as it will help me address the problems more effectively

jafingerhut commented:
> > I see multiple CI failures for your PR.
> 
> I'm not sure about the exact cause of the CI failures at the moment. Could you please provide more details about the specific failures? I'd be happy to investigate further and work on resolving these issues. Any additional information or logs you can share as it will help me address the problems more effectively

I am pretty sure that the detailed log output of all CI tests, whether they are failing or successful, are publicly available to anyone.

If you look at this PR on github.com at this link https://github.com/p4lang/behavioral-model/pull/1266 and scroll down to the bottom, near the bottom you should see a list of tests with names like these, and a "Details" link to the right of each one:

+ Test / test (clang++-10, clang-10) (pull_request).  Failing after 12m
+ Test / test-ubuntu22 (pull_request) Failing after 1m

as well as some tests that have been successful.

OsamaRab3 commented:
@antoninbas Are there any additional changes or updates that need?

antoninbas commented:
@OsamaRab3 At least one commit is not signed, so the DCO check is failing. I'd suggest squashing all 4 commits, and signing the resulting commit.

OsamaRab3 commented:
@antoninbas I reviewed the DCO check failure details and noticed that it was due to missing "Signed-off-by" lines in the commit messages. Following the instructions provided in the error message, I have rebased the  branch and added the necessary "Signed-off-by" lines to each commit.
Please let me know if there are any additional steps I should take or if further adjustments are needed

antoninbas commented on m4/ax_cxx_compile_stdcxx_17.m4:
```diff
@@ -0,0 +1,35 @@
+#serial 2
+
+AX_REQUIRE_DEFINED([AX_CXX_COMPILE_STDCXX])
+AC_DEFUN([AX_CXX_COMPILE_STDCXX_17], [AX_CXX_COMPILE_STDCXX([17], [$1], [$2])])
```
missing newline character

antoninbas commented on targets/simple_switch_grpc/m4/ax_cxx_compile_stdcxx_17.m4:
```diff
@@ -0,0 +1,35 @@
+#serial 2
+
+AX_REQUIRE_DEFINED([AX_CXX_COMPILE_STDCXX])
+AC_DEFUN([AX_CXX_COMPILE_STDCXX_17], [AX_CXX_COMPILE_STDCXX([17], [$1], [$2])])
```
ditto

OsamaRab3 commented:
@antoninbas any thing else?
